mewarpx/mewarpx/emission.py

```diff
@@ -125,36 +126,34 @@ def record_injectedparticles(self, species, E_total=None, w=None,
             argument if no particles are being added by this processor.
 
         Arguments:
-            species: [TODO - adapt to WarpX]
-            E_total (np.ndarray): Array of length npart with E_total values.
-            w (np.ndarray): Array of length ``npart`` with variable weights.
+            species (mepicmi.Species): Species of particle
+            w (np.ndarray or float): Array of length npart with particle weights.
+            E_total (np.ndarray or float): Array of length npart with E_total values.
             n (int): Number of macroparticles, _only_ needed if overriding the
                 length of E_total. This is useful mostly in the case that
                 E_total is already summed over particles, in which case a
                 single number can be passed for it rather than an array.
         """
-        raise NotImplementedError
-        # data = np.zeros(7)
 
-        # data[0] = warp.top.it*warp.top.dt
-        # data[1] = warp.top.it
-        # data[2] = species.jslist[0]
-        # data[3] = self.getvoltage_e()
+        if n is not None and np.size(w) != 1:
+            raise RuntimeError("Cannot pass array for w and specify n")
+
+        if n is None and np.size(w) == 1:
+            raise RuntimeError("Cannot pass single value for w and not specify n")
 
-        # if E_total is not None:
-        #     if n is not None:
-        #         data[4] = n
-        #     else:
-        #         data[4] = len(E_total)
+        data = np.zeros(7)
 
-        #     if w is not None:
-        #         data[5] = species.sq * species.sw * np.sum(w)
-        #     else:
-        #         data[5] = species.sq * species.sw * data[4]
+        data[0] = mwxrun.get_it() * mwxrun.get_dt()
```

Comment:
Please add a comment for each of these components describing what it represents. I can readily figure out what some of these calculations mean, but not all of them.